### ISSUE TITLE
pitivi runtime error on master

### ISSUE THREAD

**Issue by bjornfor** [NixOS/nixpkgs#15812]
## Issue description
pitivi segfaults on startup. Well, it actually manages to show the "Welcome" window, but when clicking past that it aborts with a segfault.

```
$ PYTHONPATH= ./result/bin/pitivi 
Missing soft dependency:
- pycanberra not found on the system
    -> enables sound notifications when rendering is complete
Missing soft dependency:
- GnomeDesktop not found on the system
    -> file thumbnails provided by GNOME's thumbnailers
Missing soft dependency:
- Notify not found on the system
    -> enables visual notifications when rendering is complete
Traceback (most recent call last):
  File "/nix/store/bvxl6d2wjs8v7rcxkyicj84zawzg3yif-pitivi-0.95/lib/pitivi/python/pitivi/titleeditor.py", line 394, in tabSwitchedCb
    self._connect_signals()
  File "/nix/store/bvxl6d2wjs8v7rcxkyicj84zawzg3yif-pitivi-0.95/lib/pitivi/python/pitivi/titleeditor.py", line 326, in _connect_signals
    self.app.gui.viewer.target.connect(
AttributeError: 'NoneType' object has no attribute 'viewer'

(.pitivi-wrapped:23820): GLib-GIO-CRITICAL **: g_action_print_detailed_name: assertion 'g_action_name_is_valid (action_name)' failed

(.pitivi-wrapped:23820): Gtk-CRITICAL **: gtk_application_set_accels_for_action: assertion 'detailed_action_name != NULL' failed

(.pitivi-wrapped:23820): GLib-GIO-CRITICAL **: g_action_print_detailed_name: assertion 'g_action_name_is_valid (action_name)' failed

(.pitivi-wrapped:23820): Gtk-CRITICAL **: gtk_application_set_accels_for_action: assertion 'detailed_action_name != NULL' failed

(.pitivi-wrapped:23820): GLib-GIO-CRITICAL **: g_action_print_detailed_name: assertion 'g_action_name_is_valid (action_name)' failed

(.pitivi-wrapped:23820): Gtk-CRITICAL **: gtk_application_set_accels_for_action: assertion 'detailed_action_name != NULL' failed

(.pitivi-wrapped:23820): GLib-GIO-CRITICAL **: g_action_print_detailed_name: assertion 'g_action_name_is_valid (action_name)' failed

(.pitivi-wrapped:23820): Gtk-CRITICAL **: gtk_application_set_accels_for_action: assertion 'detailed_action_name != NULL' failed

###
### At this point the pitivi GUI is up, and the "welcome" window/dialog is shown.
### I close the dialow and the below segfault happens....
###

(.pitivi-wrapped:23820): Gtk-WARNING **: Allocating size to pitivi+viewer+TransformationBox 0x3d922f0 without calling gtk_widget_get_preferred_width/height(). How does the code know the size to allocate?
Segmentation fault
```

### Steps to reproduce
On git master (51f479465c22de490fe0e55ae6ae41cade6f771a):
```
$ nix-build -A pitivi
$ PYTHONPATH= ./result/bin/pitivi
### ... and close the "welcome" dialog to witness pitivi dying due to segfault.
```

## Technical details

* System: NixOS 16.09.git.15072e8 (Flounder)

**Comment by bjornfor** [comment]
There is a Debian issue about this: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=819291

**Comment by bjornfor** [comment]
Apparently an issue with gtk 3.20. The dev version of pitivi is supposed to have a fix.

**Comment by bjornfor** [comment]
I tried updating to latest dev version (0.95-430), but it has grown a new (hard) dependency:

```
$ PYTHONPATH= ./result/bin/pitivi 
ERROR - The following hard dependencies are unmet:
==================================================
- GstTranscoder not found on the system
...
```

**Comment by bjornfor** [comment]
FYI, I asked if upstream could make a new release: https://sourceforge.net/p/pitivi/mailman/pitivi-pitivi/thread/CAEYzJUF2_4Fp1%3DqUAAQFST0N-%3DjX%3DCfEW%2B8CHVsx5gMb0Cx8eg%40mail.gmail.com/#msg35123823

**Comment by bjornfor** [comment]
So the pitivi 0.96 release is out, it should fix the gtk-3.20 incompatibility. Unfortunately, pitivi grew another dependency, gst-transcoder (a package written by pitivi developers, AFAICS). gst-transcoder uses the meson (+ ninja) build system and my first build attempts fail:

```
Traceback (most recent call last):
  File "/nix/store/fcspzg0k86dfl9x6mkav4y0q6i736q1w-meson-0.26.0/bin/meson", line 208, in run
    app.generate()
  File "/nix/store/fcspzg0k86dfl9x6mkav4y0q6i736q1w-meson-0.26.0/bin/meson", line 149, in generate
    intr.run()
  File "/nix/store/fcspzg0k86dfl9x6mkav4y0q6i736q1w-meson-0.26.0/share/meson/interpreter.py", line 954, in run
    self.evaluate_codeblock(self.ast)
  File "/nix/store/fcspzg0k86dfl9x6mkav4y0q6i736q1w-meson-0.26.0/share/meson/interpreter.py", line 976, in evaluate_codeblock
    raise e
  File "/nix/store/fcspzg0k86dfl9x6mkav4y0q6i736q1w-meson-0.26.0/share/meson/interpreter.py", line 970, in evaluate_codeblock
    self.evaluate_statement(cur)
  File "/nix/store/fcspzg0k86dfl9x6mkav4y0q6i736q1w-meson-0.26.0/share/meson/interpreter.py", line 1041, in evaluate_statement
    return self.assignment(cur)
  File "/nix/store/fcspzg0k86dfl9x6mkav4y0q6i736q1w-meson-0.26.0/share/meson/interpreter.py", line 1766, in assignment
    value = self.evaluate_statement(node.value)
  File "/nix/store/fcspzg0k86dfl9x6mkav4y0q6i736q1w-meson-0.26.0/share/meson/interpreter.py", line 1039, in evaluate_statement
    return self.function_call(cur)
  File "/nix/store/fcspzg0k86dfl9x6mkav4y0q6i736q1w-meson-0.26.0/share/meson/interpreter.py", line 1747, in function_call
    return self.funcs[func_name](node, self.flatten(posargs), kwargs)
  File "/nix/store/fcspzg0k86dfl9x6mkav4y0q6i736q1w-meson-0.26.0/share/meson/interpreter.py", line 67, in wrapped
    return f(self, node, args, kwargs)
  File "/nix/store/fcspzg0k86dfl9x6mkav4y0q6i736q1w-meson-0.26.0/share/meson/interpreter.py", line 59, in wrapped
    return f(self, node, args, kwargs)
  File "/nix/store/fcspzg0k86dfl9x6mkav4y0q6i736q1w-meson-0.26.0/share/meson/interpreter.py", line 1000, in func_import
    module = importlib.import_module('modules.' + modname).initialize()
  File "/nix/store/dbrlplrwxf21yhqymhcdr3l0r9yry9x5-python3-3.5.1/lib/python3.5/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 986, in _gcd_import
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 956, in _find_and_load_unlocked
ImportError: No module named 'modules.pkgconfig'
```

I ran some sanity checks on meson by following the official "getting started" tutorial: https://github.com/mesonbuild/meson/wiki/Tutorial. It seems to work fine at that. I'm afraid that there is a python dependency that leaks in and messes up the gst-transcoder build. Or something else. I just don't know yet.

My current WIP branch:
https://github.com/bjornfor/nixpkgs/tree/pitivi-0.96

**Comment by rasendubi** [comment]
@bjornfor the issue is caused by meson being too old and not having `modules.pkgconfig`. I've updated meson at https://github.com/NixOS/nixpkgs/pull/17217.

After that update, your pitivi builds, but fails to run with:
```
ERROR - The following hard dependencies are unmet:
==================================================
- Gst version 1.8.1.0 is installed but Pitivi requires at least version 1.8.2
- GES version 1.8.1.0 is installed but Pitivi requires at least version 1.8.2
- gstgtk version 1.8.1 is installed but Pitivi requires at least version 1.8.2
Missing soft dependency:
- pycanberra not found on the system
    -> enables sound notifications when rendering is complete
```

**Comment by rasendubi** [comment]
Well. These dependencies were updated in master. Rebasing on it fixes the pitivi.

@bjornfor, would you like to open a PR with your changes?

**Comment by bjornfor** [comment]
@rasendubi: Thanks for finding the reason for the meson build issue!

Here is my PR for upgrading and fixing pitivi: https://github.com/NixOS/nixpkgs/pull/17228

**Comment by bjornfor** [comment]
Fixed in master as of 245168060372cf84ce34fae8af6402b948d33689.
